### PR TITLE
improve(LM rewards): YD-ETH-MAR21 week 35 liquidity mining rewards (includes roll)

### PR DESCRIPTION
**Pull request by chrismaree**
This PR adds week 35's liquidity mining payouts for YD-ETH-MAR21 rolled with YD-WETH-JUN21. The scripts was run from 2021-03-22 23:00:00 UTC (12091493) to 2021-03-30 23:00:00 UTC (12136988).

**Note** that the payouts for this week's rewards are based on the UMA Dev mining program. Based off the dev mining program, half of all rewards allocated to the YD-ETH-MAR21 contract, pro-rata to other contracts within the UMA dev mining program, are allocated as liquidity mining rewards.

https://github.com/UMAprotocol/protocol/pull/2787/files

This week contained a roll for Liquidity mining. To accommodate this, the expiring token is considered the rolled _from_ contract and the new contract is considered the rolled _to_ contract. Normally this kind of roll would be done using the `CalculateRollBalancerLpRewards.js` script. However, this script was design to work with a fixed number of uma to split between two contracts during the roll. We are now fortunate enough to use developer mining to correctly attribute between the two contracts.

To deal with this without writing any extra scripts we can simply run the normal `CalculateBalancerLPRewards` script twice and move the first output to be a rolled input. then, use the `JoinRolledPayouts` script to construct a joined payout file containing the sum of the recipients.

From this week's dev mining, a total of `8643.929416342063317219` UMA is allocated to the YD-ETH-MAR21 contract. As UMA has decided to send half of these rewards as LM rewards, the total allocated for these weekly payouts is 4321 UMA for YD-ETH-MAR21.

Similarly, this week sees YD-ETH-JUN21 earning `2420.627914622850645558` UMA. With half of this attributed this works out to 1210. 

**To re-generate these results run:**

1. Run the `CalculateBalancerLPRewards` to produce the expiring week rewards.
```
node ./liquidity-mining/CalculateBalancerLPRewards.js --fromBlock 12091493 --toBlock 12136988 --poolAddress="0x5e065d534d1daaf9e6222afa1d09e7dac6cbd0f7" --umaPerWeek=4321 --tokenName="yusdeth" --week=35 --network mainnet_mnemonic
```
2. To deal with the roll, we move this file to a new roll file to store the information for verification. Note that this is different to how previous rolls are done as we are accommodating the dev mining inputs.

```
mv ./liquidity-mining/yusdeth-weekly-payouts/Week_35_Mining_Rewards.json ./liquidity-mining/yusdeth-weekly-payouts/expiring-contract-rolls/Roll_4_Mining_Rewards.json
```
3. Generate the LM rewards for the new contract.
```
node ./liquidity-mining/CalculateBalancerLPRewards.js --fromBlock 12091493 --toBlock 12136988 --poolAddress="0xf08241998394D61dC0bbbAC767B0B8242549761F" --umaPerWeek=1210 --tokenName="yusdeth" --week=35 --network mainnet_mnemonic
```

4. Join the payout files to create one condensed output

```
node ./liquidity-mining/JoinRolledPayouts.js --week 35 --rollNum 4 --tokenName yusdeth
```

**Files produced**
Due to the rolling action, two files are produced. **The canonical payout that should be considered the sum of the two contracts is the `shareHolderPayout` within `packages/affiliates/liquidity-mining/yusdeth-weekly-payouts/Week_35_Mining_Rewards.json`**

### CODE REVIEW

File: packages/affiliates/liquidity-mining/yusdeth-weekly-payouts/Week_35_Mining_Rewards.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
+{
+  "week": 35,
+  "fromBlock": 12091493,
+  "toBlock": 12136988,
+  "poolAddress": ["0x5e065d534d1daaf9e6222afa1d09e7dac6cbd0f7", "0xf08241998394D61dC0bbbAC767B0B8242549761F"],
+  "blocksPerSnapshot": 256,
+  "umaPerWeek": 5531,
+  "shareHolderPayout": {
```
Comment (chrismaree) [inline packages/affiliates/liquidity-mining/yusdeth-weekly-payouts/Week_35_Mining_Rewards.json:8]: this `shareHolderPayout` contains the sum of the roll between the two contracts.👇

**Comment by smb2796** [comment]
It looks like the block numbers you used correspond to the correct timestamps, but the dates noted in your description are off. Everything else looks good to me though

**Comment by chrismaree** [comment]
> It looks like the block numbers you used correspond to the correct timestamps, but the dates noted in your description are off. Everything else looks good to me though

good catch! have updated. thanks for laser eyes!

**Review by hlambur — APPROVED** [review]
LGTM!

**Comment by hlambur** [comment]
Sent with tx: https://etherscan.io/tx/0x6c5726e6fbdc09599c568b1db7825912b113d1e14f19ec41b38e7bff06e34b3c

**Pull request merged**